Multiple fixes to 'docker stats' output

ottok opened the pull request.

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "master" git@github.com:ottok/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~3
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

duglin commented:
please squash your commits down to one

ottok commented:
Failing test is not my fault..
```
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --homedir /tmp/tmp.cAYPC6N3Ut --no-auto-check-trustdb --trust-model always --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys E871F18B51E0147C77796AC81196BA81F6B0FC61
gpg: requesting key F6B0FC61 from hkp server p80.pool.sks-keyservers.net
?: p80.pool.sks-keyservers.net: Host not found
gpgkeys: HTTP fetch error 7: couldn't connect: Connection timed out
```

aboch commented on api/client/stats.go:
```diff
@@ -71,8 +71,12 @@ func (s *containerStats) Collect(cli *DockerCli, streamStats bool) {
 			s.MemoryPercentage = memPercent
-			s.NetworkRx = float64(v.Network.RxBytes)
-			s.NetworkTx = float64(v.Network.TxBytes)
+			// Note: The cgroup network Rx and Tx are flipped compared to ifconfig
```
@ottok I might be wrong here and not be fully aware of the context, but please check the changes in #14324 and see if it is still needed to swap rx/tx counters. 
Docker 1.7.1 now uses libnetwork to retrieve the container interface stats and libnetwork reports them as they would appear in the ifconfig o/p on the container.

ottok replied:
I am running Docker 1.7.1 and I see that the Rx/Tx is in wrong order. If you are in doubt of this change and maybe plan to fix it on a lower level (stats API) I can clean out this change from the PR so that you can accept the rest of the PR.

ottok committed:
Multiple fixes to 'docker stats' output:

* Add space between values in docker stats output for easier parsing

  Old output could not be parsed easily because there were columns
  that did not have any separator. Also values that are together
  without any space is difficult to read even for humans.

* Update unit.HumanSize comment to match what the does actually does

Signed-off-by: Otto Kekäläinen <otto@seravo.fi>

ottok commented:
@aboch I removed the Tx/Rx flipping part as that seemed to stall merging the rest ot the pull request. I also rebased the pull request on latest upstream master. Please pull.

tiborvass commented on pkg/units/size.go:
```diff
@@ -49,7 +49,7 @@ func CustomSize(format string, size float64, base float64, _map []string) string
 
 // HumanSize returns a human-readable approximation of a size
-// using SI standard (eg. "44kB", "17MB").
+// capped at 4 valid numers (eg. "2.746 MB", "796 KB").
```
`numbers`

ottok replied:
@tiborvass I'll push a typofix in a minute, thanks for spotting this!

tiborvass commented:
LGTM bar small nit

LK4D4 commented:
LGTM
